subsys/net/lib/http/http_server_http2.c

```diff
@@ -869,6 +971,12 @@ int handle_http1_to_http2_upgrade(struct http_client_ctx *client)
 			if (ret < 0) {
 				goto error;
 			}
+		} else if (detail->type == HTTP_RESOURCE_TYPE_STATIC_FS) {
+			ret = handle_http2_static_fs_resource(
```

Comment:
It should also be supported from `handle_http_frame_headers()`, this handles only HTTP upgrade case.